Automattic/wp-calypso
# Email verification: send using data layer

This PR updates the way we send email verification requests to use redux actions. The component I updated so far is `VerifyEmailDialog`, and after it's reviewed I'll also update these components in subsequent PRs:
- [ ] `EmailUnverifiedNotice`
- [ ] `HelpUnverifiedWarning`

## Testing Instructions
You can test this PR by making sure that the verify email dialog works as expected on the posting page as an unverified user:
1. Log in to WordPress.com as an unverified user
2. Go to the post edit page: calypso.localhost:3000/post/yoursite.wordpress.com
3. Click the 'Learn More' link that appears below the publish button, and make sure the dialog works as expected:
<img width="346" alt="screen shot 2017-04-30 at 5 51 50 pm" src="https://cloud.githubusercontent.com/assets/11487924/25568310/d3db1192-2dcd-11e7-821c-e1cdf4cd87c6.png">
